x-pack/solutions/observability/plugins/apm/public/components/shared/environment_badge/index.tsx

```diff
@@ -9,13 +9,20 @@ import React from 'react';
 import { i18n } from '@kbn/i18n';
 import { ItemsBadge } from '../item_badge';
 import { NotAvailableEnvironment } from '../not_available_popover/not_available_environment';
+import {
+  ENVIRONMENT_NOT_DEFINED_VALUE,
+  getEnvironmentLabel,
+} from '../../../../common/environment_filter_values';
 
 interface Props {
   environments: string[];
   isMetricsSignalType?: boolean;
 }
 
 export function EnvironmentBadge({ environments = [], isMetricsSignalType = true }: Props) {
+  environments = environments.length
+    ? environments
+    : [getEnvironmentLabel(ENVIRONMENT_NOT_DEFINED_VALUE)];
```

Comment:
can we avoid mutating the argument received?
```suggestion
  const _environments = environments.length
    ? environments
    : [getEnvironmentLabel(ENVIRONMENT_NOT_DEFINED_VALUE)];
```

Use `_environments` or any other name.